client/src/main/java/io/pravega/client/netty/impl/FlowHandler.java

```diff
@@ -313,14 +318,28 @@ public void close() {
     }
 
     private final class KeepAliveTask implements Runnable {
+        
+        private final ChannelFutureListener listener = new ChannelFutureListener() {
+            @Override
+            public void operationComplete(ChannelFuture future) throws Exception {
+                recentMessage.set(true);
+                if (!future.isSuccess()) {
+                    future.channel().pipeline().fireExceptionCaught(future.cause());
```

Comment:
Can we log indicating that sending a keep-alive failed here?

Reply:
Done